src/main/java/org/folio/rest/impl/OaiPmhImpl.java

```diff
@@ -0,0 +1,146 @@
+    oai.getRequest()
+       .withResumptionToken(resumptionToken)
+       //.withFrom(from != null ? Instant.parse(from) : null)
+       //.withUntil(until != null ? Instant.parse(until) : null)
+       .withSet(set)
+       .withMetadataPrefix(metadataPrefix);
+
+    try {
+      String response = ResponseHelper.getInstance().writeToString(oai);
+      asyncResultHandler.handle(succeededFuture(GetOaiRecordsResponse.respond422WithApplicationXml(response)));
+    } catch (Exception e) {
+      logger.error("Unexpected error happened while processing ListRecords verb request", e);
+      asyncResultHandler.handle(succeededFuture(GetOaiRecordsResponse.respond500WithTextPlain(getErrorMessage())));
+    }
+  }
+
+  @Override
+  public void getOaiRecordsById(String id, String metadataPrefix, Map<String, String> okapiHeaders,
+                                Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    OAIPMH oai = buildBaseResponse(VerbType.GET_RECORD)
+      .withErrors(objectFactory.createOAIPMHerrorType().withCode(OAIPMHerrorcodeType.ID_DOES_NOT_EXIST));
+    oai.getRequest().withIdentifier(id).withMetadataPrefix(metadataPrefix);
+
+    try {
+      String response = ResponseHelper.getInstance().writeToString(oai);
+      asyncResultHandler.handle(succeededFuture(GetOaiRecordsByIdResponse.respond422WithApplicationXml(response)));
+    } catch (Exception e) {
+      logger.error("Unexpected error happened while processing GetRecord verb request", e);
+      asyncResultHandler.handle(succeededFuture(GetOaiRecordsByIdResponse.respond500WithTextPlain(getErrorMessage())));
+    }
+  }
+
+  @Override
+  public void getOaiIdentifiers(String resumptionToken, String from, String until, String set, String metadataPrefix,
+                                Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+                                Context vertxContext) {
+    asyncResultHandler.handle(succeededFuture(GetOaiIdentifiersResponse.respond500WithTextPlain("The verb ListIdentifiers is not supported yet :(")));
+  }
+
+  @Override
+  public void getOaiMetadataFormats(String identifier, Map<String, String> okapiHeaders,
+                                    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    asyncResultHandler.handle(succeededFuture(GetOaiMetadataFormatsResponse.respond500WithTextPlain("The verb ListMetadataFormats is not supported yet :(")));
+  }
+
+  @Override
+  public void getOaiSets(String resumptionToken, Map<String, String> okapiHeaders,
+                         Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    asyncResultHandler.handle(succeededFuture(GetOaiSetsResponse.respond500WithTextPlain("The verb ListSets is not supported yet :(")));
+  }
+
+  @Override
+  public void getOaiRepositoryInfo(Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
```

Comment:
Consider splitting the implementation of each endpoint out into its own class.  Otherwise this class will get very long as we add implementations for the other endpoints.

Reply:
I agree, it's a good point.
Done - the repository_info endpoint implementation was split into its own class.